src/apps/allbridge/positions.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  PositionsHook,
+  AppTokenPositionDefinition,
+  UnknownAppTokenError,
+  TokenDefinition,
+  ContractPositionDefinition,
+} from '../../types/positions'
+import { Address } from 'viem'
+import { DecimalNumber, toDecimalNumber } from '../../types/numbers'
+import BigNumber from 'bignumber.js'
+import { getClient } from '../../runtime/client'
+import { getTokenId } from '../../runtime/getTokenId'
+import { getAllbridgeTokenInfo } from './api'
+import { poolAbi } from './abis/pool'
+
+const ALLBRIDGE_LOGO =
+  'https://raw.githubusercontent.com/valora-inc/dapp-list/main/assets/allbridgecore.png'
+
+const hook: PositionsHook = {
+  getInfo() {
+    return {
+      id: 'allbridge',
+      name: 'Allbridge',
+      description: '',
+    }
+  },
+  async getPositionDefinitions(networkId, address) {
+    const allbridgeTokenInfo = (await getAllbridgeTokenInfo({ networkId }))
+      .tokens
+
+    const client = getClient(networkId)
+
+    const [balances, rewards] = await Promise.all([
+      await Promise.all(
+        allbridgeTokenInfo.map(async (tokenInfo) => {
+          return address
+            ? client.readContract({
+                address: tokenInfo.poolAddress,
+                abi: poolAbi,
+                functionName: 'balanceOf',
+                args: [address as Address],
+              })
+            : undefined
+        }),
+      ),
+      await Promise.all(
+        allbridgeTokenInfo.map(async (tokenInfo) => {
+          return address
+            ? client.readContract({
+                address: tokenInfo.poolAddress,
+                abi: poolAbi,
+                functionName: 'pendingReward',
+                args: [address as Address],
+              })
+            : undefined
+        }),
+      ),
+    ])
+
+    return allbridgeTokenInfo.flatMap((tokenInfo, i) => {
+      const apr = new BigNumber(tokenInfo.apr7d).toNumber() * 100
+
+      const balanceOf = balances[i]
+      const pendingReward = rewards[i]
+
+      const showLpToken = !address || (!!balanceOf && balanceOf > 0)
+      const showReward = !!pendingReward && pendingReward > 0
```

Comment:
```suggestion
      const showLpToken = !address || balanceOf > 0
      const showReward = pendingReward > 0
```
the `!!` seems rendundant

Reply:
without this I get a typescript error, since `balanceOf` and `pendingReward` are possibly undefined

Reply:
it's undefined because of the logic I have for the readContract calls, where if there is not an address it just sets them to undefined.